app/controllers/sidebar_sections_controller.rb

```diff
@@ -35,6 +35,18 @@ def update
     ActiveRecord::Base.transaction do
       sidebar_section.update!(section_params.merge(sidebar_urls_attributes: links_params))
       sidebar_section.sidebar_section_links.update_all(user_id: sidebar_section.user_id)
+      order =
+        sidebar_section
+          .sidebar_urls
+          .sort_by do |url|
+            links_params.index { |link| link["name"] == url.name && link["value"] == url.value } ||
+              -1
+          end
```

Comment:
Just to confirm, when the url exists in the `links` param, we want to sort it based on the order it appears in the `links` params but when a particular url does not exists it should always be sorted above the urls that exists in the `link` params?

Reply:
Yes, this is a fallback when request for example is incomplete and not containing all links. One admin can add a link and another may make an update request. Quite rare, but don't want to error in that case.